bot/exts/help_channels/_cog.py

```diff
@@ -465,3 +484,32 @@ async def wait_for_dormant_channel(self) -> discord.TextChannel:
         self.queue_tasks.remove(task)
 
         return channel
+
+    async def update_available_help_channels(self) -> None:
+        """Updates the dynamic message within #how-to-get-help for available help channels."""
+        if not self.available_help_channels:
+            self.available_help_channels = set(
+                c for c in self.available_category.channels if not _channel.is_excluded_channel(c)
+            )
+
+        available_channels = AVAILABLE_HELP_CHANNELS.format(
+            available=', '.join(c.mention for c in self.available_help_channels) or None
+        )
+
+        if self.dynamic_message is not None:
+            try:
+                await self.bot.http.edit_message(
+                    constants.Channels.how_to_get_help, self.dynamic_message, content=available_channels
+                )
+                log.trace("Help channels have changed, dynamic message has been edited.")
+            except discord.NotFound:
+                pass
+            else:
+                return
+
+        log.trace("No How-to-get-help dynamic message could be found in the Redis cache. Setting a new one.")
```

Comment:
This log is misleading; we hit this line either in the described case or when the message ID was there but the edit failed.